app/src/main/java/com/github/swent/echo/compose/components/EventInfoSheet.kt

```diff
@@ -0,0 +1,195 @@
+package com.github.swent.echo.compose.components
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Face
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.ModalBottomSheet
+import androidx.compose.material3.Text
+import androidx.compose.material3.rememberModalBottomSheetState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+
+/**
+ * Bottom sheet that shows the details of an event.
+ *
+ * @param showBottomSheet state to show the bottom sheet (mutable)
+ * @param eventTitle title of the event
+ * @param eventOrganization organization hosting the event
+ * @param eventDescription description of the event
+ * @param eventDateTime date and time of the event
+ * @param eventImage image id of the event
+ * @param eventPeople number of people who joined the event
+ * @param eventPeopleMax maximum number of people who can join the event
+ * @param onJoinButtonPressed callback to join the event
+ * @param onShowPeopleButtonPressed callback to show people who joined the event
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun EventInfoSheet(
+    showBottomSheet: MutableState<Boolean>,
+    eventTitle: String,
+    eventOrganization: String,
+    eventDescription: String,
+    eventDateTime: String,
+    eventImage: Int, // not sure how to handle images in the data system yet
+    eventPeople: Int,
+    eventPeopleMax: Int,
+    onJoinButtonPressed: () -> Unit,
+    onShowPeopleButtonPressed: () -> Unit
+) {
+    val sheetState = rememberModalBottomSheetState()
```

Comment:
You can pass a `confirmValueChange` parameter to `rememberModalBottomSheetState()`, which is a callback you can use to determine if the bottom sheet is in the expanded state.

```kotlin
val sheetState = rememberModalBottomSheetState(skipPartiallyExpanded = false) {value ->
    if (value == SheetValue.Expanded) {
        // Call the onFullyExpanded callback
    }
    true
}
```

Reply:
I'll try that !